python/TestHarness/schedulers/RunParallel.py

```diff
@@ -14,30 +14,30 @@ def validParams():
     def __init__(self, harness, params):
         Scheduler.__init__(self, harness, params)
 
-    def run(self, job_container):
+    def run(self, job):
         """ Run a tester command """
-        tester = job_container.getTester()
+        tester = job.getTester()
 
         # Launch and wait for the command to finish
-        job_container.run()
+        job.run()
 
         # Was this test already considered finished? (Timeouts, Dry Run)
         if tester.isFinished():
             return
 
-        output = job_container.getOutput()
+        output = job.getOutput()
 
         # If we are doing recover tests
         if self.options.enable_recover and tester.specs.isValid('skip_checks') and tester.specs['skip_checks']:
-            tester.setStatus('PART1', tester.bucket_success)
+            self.adjustJobStatus(job, tester.bucket_success, 'PART1')
```

Comment:
OK - so we dropped setStatus?

Reply:
I am using a method in the Scheduler base class to set all statuses now. adjustJobStatus sets the tester's status in a thread lock state.